airbyte-integrations/connectors/source-zuora/source_zuora/zuora_auth.py

```diff
@@ -0,0 +1,63 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+from typing import Dict
+
+import requests
+
+
+class ZuoraAuthenticator:
+    def __init__(self, is_sandbox: bool = True):
+        self.is_sandbox = is_sandbox
+
+    @property
+    def endpoint(self) -> str:
+        if self.is_sandbox:
+            return "https://rest.apisandbox.zuora.com"
+        else:
+            return "https://rest.zuora.com"
+
+    # GENERATE TOKEN
+    def generateToken(self, client_id: str, client_secret: str) -> Dict:
+        endpoint = f"{self.endpoint}/oauth/token"
+        header = {"Content-Type": "application/x-www-form-urlencoded"}
+        data = {
+            "client_id": f"{client_id}",
+            "client_secret": f"{client_secret}",
+            "grant_type": "client_credentials",
+        }
+        try:
+            session = requests.post(endpoint, headers=header, data=data)
```

Comment:
Does not it provide expiration time as well?

Reply:
According to the section of [documentation](https://www.zuora.com/developer/api-reference/#section/Authentication/OAuth-v2.0): `Other Supported Authentication Schemes` 
It's adjustable in the Administration setting of account:
"Administration > Security Policies > Session timeout" so it could be different for a different accounts, there is no information inside the response after Generating Token about the live period of the token.

Reply:
we should read this "expires_in": 3599  used it  to   refresh token  before it is expired

Reply:
please  inherit from CDK's authenticator